src/Playwright/Core/BrowserType.cs

```diff
@@ -117,5 +118,24 @@ public async Task<IBrowserContext> LaunchPersistentContextAsync(string userDataD
                 recordHarOmitContent: options.RecordHarOmitContent,
                 recordVideo: Browser.GetVideoArgs(options.RecordVideoDir, options.RecordVideoSize)).ConfigureAwait(false)).Object;
         }
+
+        public async Task<IBrowser> ConnectAsync(string wsEndpoint, BrowserTypeConnectOptions options = null)
+        {
+            options ??= new BrowserTypeConnectOptions();
+#pragma warning disable CA2000 // Dispose objects before losing scope
```

Comment:
I can remove it but ensuring that WebSocket objects & process objects get cleaned up. Ideally, we should keep it. thoughts? or any suggestion on refactoring?

Reply:
Also, websocket contains dispose method that needs to be disposed of. This is after removing `IDisposable`
![image](https://user-images.githubusercontent.com/9930610/125572023-185e75be-a2f5-44e8-a07d-45db83208db1.png)

Reply:
I think using `IDisposable` is the lesser of the two evils, because it gives us more good than bad. Yes, it's annoying having to disable the one warning, but, also, it's the recommended practice: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000#when-to-suppress-warnings